tests/core/test_expectation_configuration.py

```diff
@@ -172,3 +172,30 @@ def test_expectation_configuration_get_evaluation_parameter_dependencies_with_qu
     # Should fully skip `nested_update` calls in method due to lacking an "expectation_suite_name" key
     dependencies = ec.get_evaluation_parameter_dependencies()
     assert dependencies == {}
+
+
+@pytest.mark.unit
+def test_expectation_configuration_to_domain_obj():
```

Comment:
I know we hate updating the ExpectationConfiguration, but while you're here, I'd probably just parameterize this test to go through notes being string, list, none.